experiments/sample_ssl4eo.py

```diff
@@ -0,0 +1,265 @@
+        url = "https://simplemaps.com/static/data/world-cities/basic/simplemaps_worldcities_basicv1.71.zip"  # noqa: E501
+        filename = "worldcities.csv"
+        if not os.path.exists(os.path.join(download_root, os.path.basename(url))):
+            download_and_extract_archive(url, download_root)
+        with open(os.path.join(download_root, filename), encoding="UTF-8") as csvfile:
+            reader = csv.DictReader(csvfile, delimiter=",", quotechar='"')
+            cities = []
+            for row in reader:
+                row["population"] = (
+                    row["population"].replace(".", "") if row["population"] else "0"
+                )
+                cities.append(row)
+        return cities
+
+    @staticmethod
+    def get_interest_points(
+        cities: List[Dict[str, str]], size: int = 10000
+    ) -> List[List[float]]:
+        cities = sorted(cities, key=lambda c: int(c["population"]), reverse=True)[:size]
+        points = [[float(c["lng"]), float(c["lat"])] for c in cities]
+        return points
+
+    @staticmethod
+    def km2deg(kms: float, radius: float = 6371) -> float:
+        return kms / (2.0 * radius * np.pi / 360.0)
+
+    @staticmethod
+    def deg2km(deg: float, radius: float = 6371) -> float:
+        return deg * (2.0 * radius * np.pi / 360.0)
+
+
+class BoundedUniformSampler:
+    def __init__(self, boundaries: shape = None) -> None:
+        if boundaries is None:
+            self.boundaries = self.get_country_boundaries()
+        else:
+            self.boundaries = boundaries
+
+    def sample_point(self) -> List[float]:
+        minx, miny, maxx, maxy = self.boundaries.bounds
+        lon = np.random.uniform(minx, maxx)
+        lat = np.random.uniform(miny, maxy)
+        p = Point(lon, lat)
+        if self.boundaries.contains(p):
+            return [p.x, p.y]
+        else:
+            return self.sample_point()
+
+    @staticmethod
+    def get_country_boundaries(
+        download_root: str = os.path.expanduser("~/.cache/naturalearth"),
+    ) -> shape:
+        url = "https://www.naturalearthdata.com/http//www.naturalearthdata.com/download/110m/cultural/ne_110m_admin_0_countries.zip"  # noqa: E501
+        filename = "ne_110m_admin_0_countries.shp"
+        if not os.path.exists(os.path.join(download_root, os.path.basename(url))):
+            download_and_extract_archive(url, download_root)
+        sf = shapefile.Reader(os.path.join(download_root, filename))
+        return shape(sf.shapes().__geo_interface__)
+
+
+class OverlapError(Exception):
+    pass
+
+
+def sample_random_locations_rtree(
+    idx: int,
+    sampler: Any,
+    radius: float,
+    overlap_ratio: float,
+    rtree_obj: index.Index = index.Index(),
+) -> List[float]:
+    """sample new coord and check overlap --- rtree"""
+    # use rtree to avoid strong overlap
+    count = 0
+    bbox_radius = (1 - overlap_ratio) * radius / 1000
```

Comment:
Let's not call it "radius" since we're using boxes instead of circles now. "size" is what we usually use in TorchGeo to mean width/height